api/src/domain/tax-filings/taxFilingsFactory.ts

```diff
@@ -0,0 +1,43 @@
+import { UserData } from "@shared/userData";
+import { ApiTaxFilingClient, TaxFilingClient, UserDataClient } from "../types";
+
+export const taxFilingsFactory = (
+  userDataClient: UserDataClient,
+  taxFilingClient: ApiTaxFilingClient
+): TaxFilingClient => {
+  const lookup = async (props: {
+    userId: string;
+    taxId: string;
+    businessName: string;
+  }): Promise<UserData> => {
+    let userData = await userDataClient.get(props.userId);
```

Comment:
I think we talked about this a month or so ago when we were adding operating phase to the backend, but I believe we agreed that we wanted to avoid having domain functions like this do the get/put userDataClient operations. Instead, we wanted these to be more "pure" functions where they get passed a userData, do stuff with it, make some changes, and return the updated userData, and then the caller is responsible for both the get and the put. 

This makes things easier because:
- we can chain update functions like this if need be, because they just operate on what they're passed instead of what's in the DB
- it makes the test simpler because we just pass in a userData object instead of having to mock the userDataClient in this test
- it makes this function more reusable